File: COM_backend/src/main/java/com/uos/comp6239backend/tdata/entity/ReaderStoryHistory.java
package com.uos.comp6239backend.tdata.entity;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import java.io.Serializable;

/**
 * @title: ReaderStoryHistory
 * @Author Hym
 * @Date: 2024-05-12 17:47
 * @Description: Represents the reading history of a reader for a specific story.
 * @Version 1.0
 */
@Data
@ApiModel(description = "读者故事历史实体")
@Getter
@Setter
@ToString
public class ReaderStoryHistory implements Serializable {
    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "读者ID")
    private Integer readerId;

    @ApiModelProperty(value = "故事ID")
    private Integer storyId;

    @ApiModelProperty(value = "是否使用", example = "0: 不使用, 1: 使用")
    private Integer isUsed;

    @ApiModelProperty(value = "笔记")
    private String notes;

    @ApiModelProperty(value = "当前进度")
    private Integer currentProgress;
}

File: COM_project/app/src/main/java/com/codelab/basiclayouts/model/reader/readerStoryHistory.kt
package com.codelab.basiclayouts.model.reader

import com.google.gson.annotations.SerializedName

/**
 * 读者故事历史实体类，用于跟踪读者的阅读进度和历史记录。
 */
data class readerStoryHistory(
    /** 读者ID */
    @SerializedName("readerId") val readerId: Int = -1,

    /** 故事ID */
    @SerializedName("storyId") val storyId: Int = -1,

    /** 是否正在使用 */
    @SerializedName("isUsed") val isUsed: Int = -1,

    /** 笔记 */
    @SerializedName("notes") val notes: String? = null,

    /** 当前进度 */
    @SerializedName("currentProgress") val currentProgress: Int = -1
)